### PR TITLE
re #13, re #14

### PR DESCRIPTION
**Pull request by elbosso**
This is the code I added to achieve all things needed to write an xml editor that has a XPATH search that highlights all matching regions inside the text view as seen in #14 - I thought maybe it would be interesting to others also...

### CODE REVIEW

File: src/main/java/org/codehaus/staxmate/dom/DOMConverter.java [hunk 1]
```diff
@@ -167,21 +180,40 @@ public void buildDocument(XMLStreamReader r, Document doc)
          */
         boolean wholeDoc = (r.getEventType() == XMLStreamConstants.START_DOCUMENT);
 
+		nsmap=new java.util.HashMap();
+
         XMLStreamReader2 sr = Stax2ReaderAdapter.wrapIfNecessary(r);
         QNameRecycler recycler = new QNameRecycler();
         boolean nsAware = _isNamespaceAware(sr);
         Node current = doc; // At top level
+		int position=0;
```
Comment (cowtowncoder) [inline src/main/java/org/codehaus/staxmate/dom/DOMConverter.java:189]: One quick note: indentation is off, maybe it's using tabs? If so, should convert to all spaces, makes diff easier to read.

File: src/main/java/org/codehaus/staxmate/dom/DOMConverter.java [hunk 2]
```diff
@@ -167,21 +180,40 @@ public void buildDocument(XMLStreamReader r, Document doc)
...
+		nodeLocationMap=new java.util.HashMap();
```
Comment (cowtowncoder) [inline src/main/java/org/codehaus/staxmate/dom/DOMConverter.java:190]: Just use `import` for these

File: src/main/java/org/codehaus/staxmate/dom/Tupel.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+package org.codehaus.staxmate.dom;
+
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ *
+ * @author elbosso
+ */
+public class Tupel<T,U> extends java.lang.Object
```
Comment (cowtowncoder) [inline src/main/java/org/codehaus/staxmate/dom/Tupel.java:15]: Misspelling, should he `Tuple`. Also, no need to indicate `extends Object`, that's implicit.

File: src/main/java/org/codehaus/staxmate/dom/Tupel.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
+/*
```
Comment (cowtowncoder) [inline src/main/java/org/codehaus/staxmate/dom/Tupel.java:1]: Remove the template comment.

File: src/main/java/org/codehaus/staxmate/dom/Tupel.java (hunk 3 above)
Comment (cowtowncoder) [reply src/main/java/org/codehaus/staxmate/dom/Tupel.java:15]: Also: for purposes here, looks bit like overkill; maybe trim all methods not used.

**Comment by cowtowncoder** [comment]
First of all, thank you for this contribution.

I added some comments, but there are a couple of bigger questions I have...

First, I noticed there are no unit tests We would need something to verify that code actually works to basic degree. It would also show something about functionality added (which new things can be done).

Second: it would be good to a proper description of added functionality (it sounds like `Location` information will be available; and partial namespace lookup) on this PR and/or related issues you filed (#13 and #14). Most users do not go through actual diffs to code.

File: src/main/java/org/codehaus/staxmate/dom/DOMConverter.java (hunk 1 above)
Comment (elbosso) [reply src/main/java/org/codehaus/staxmate/dom/DOMConverter.java:189]: i have addressed this point in my latest commit

File: src/main/java/org/codehaus/staxmate/dom/DOMConverter.java (hunk 2 above)
Comment (elbosso) [reply src/main/java/org/codehaus/staxmate/dom/DOMConverter.java:190]: addressed in last commit

File: src/main/java/org/codehaus/staxmate/dom/Tupel.java (hunk 3 above)
Comment (elbosso) [reply src/main/java/org/codehaus/staxmate/dom/Tupel.java:15]: sorry for the typo - shows i'm a German...
I left it in for the time being

**Comment by elbosso** [comment]
added tests and needed resources with the last commit

**Pull request closed without merging**